server/src/main/java/com/objectcomputing/checkins/services/team/member/TeamMemberServicesImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.objectcomputing.checkins.services.team.member;
+
+import com.objectcomputing.checkins.services.team.TeamBadArgException;
+import com.objectcomputing.checkins.services.team.TeamRepository;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileRepository;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+public class TeamMemberServicesImpl implements TeamMemberServices {
+
+    @Inject
+    private TeamRepository teamRepo;
+    @Inject
+    private TeamMemberRepository teamMemberRepo;
+    @Inject
+    private MemberProfileRepository memberRepo;
+
+    public TeamMember save(TeamMember teamMember) {
+        TeamMember teamMemberRet = null;
+        if (teamMember != null) {
+            final UUID teamId = teamMember.getTeamid();
+            final UUID memberId = teamMember.getMemberid();
+            if (teamId == null || memberId == null) {
+                throw new TeamBadArgException(String.format("Invalid teamMember %s", teamMember));
```

Comment:
These null checks won't be hit in practice because the controller has @Valid annotation. Nothing wrong with two checks, but if we want this message format (rather than "teamMember.memberid: must not be null" which the ConstraintViolationException shows) we want the controller will need an error handler for ConstraintViolationException.

Reply:
True, he copied what I originally wrote.

Since the service "could" be called from elsewhere I just threw that in there, I think mainly because Intellij was yelling at me.

I would argue just to leave it.

Reply:
If we decide to take a different approach, then lets do something similar to what I commented on Hollys PR:

> Replace this with, and make similar changes to update:
> 
> ```
>     public MemberSkill save(@NotNull MemberSkill memberSkill) {
>         MemberSkill memberSkillRet;
>         
>         @NotNull
>         final UUID memberId = memberSkill.getMemberid();
>         @NotNull
>         final UUID skillId = memberSkill.getSkillid();
> 
>         if (memberSkill.getId() != null) {
>             throw new MemberSkillBadArgException(String.format("Found unexpected id %s for member skill", memberSkill.getId()));
>         } else if (memberProfileRepository.findById(memberId).isEmpty())) {
>             throw new MemberSkillBadArgException(String.format("Member Profile %s doesn't exist", memberId));
>         } else if (skillRepository.findById(skillId).isEmpty()) {
>             throw new MemberSkillBadArgException(String.format("Skill %s doesn't exist", skillId));
>         } else if (memberSkillRepository.findByMemberidAndSkillid(memberSkill.getMemberid(),
>                 memberSkill.getSkillid()).isPresent()) {
>             throw new MemberSkillBadArgException(String.format("Member %s already has this skill %s", memberId, skillId));
>         }
> 
>         memberSkillRet = memberSkillRepository.save(memberSkill);
>         return memberSkillRet;
> 
>     }
> ```
> 
> Add `@Valdidated` to the class.
>